src/durableorchestrationclient.ts

```diff
@@ -193,15 +193,26 @@ export class DurableOrchestrationClient implements DurableClient {
             instanceId,
             ...options,
         };
-        const response = await this.getStatusInternal(internalOptions);
+        const response: AxiosResponse = await this.getStatusInternal(internalOptions);
 
         switch (response.status) {
-            case 200: // instance completed
-            case 202: // instance in progress
-            case 400: // instance failed or terminated
-            case 404: // instance not found or pending
-            case 500: // instance failed with unhandled exception
-                return response.data as DurableOrchestrationStatus;
+            case 200: // instance completed, failed or terminated
+            case 202: // instance running or pending
+            case 400:
+                if (!response.data) {
+                    throw new Error(
+                        "Operation failed. The received response contained empty response data."

```

Comment:
nit I think the following would make the error more concrete.
```suggestion
                        "DurableClient error: the Durable Functions extension replied with an empty HTTP 400 response."
```

Reply:
Do we have a repro producing an HTTP 400 response? From the internet, HTTP 400 seems to correspond to a malformed client request so I doubt there's a "real" DurableOrchestrationStatus response in here. If this is just an error, maybe we should treat this the same as HTTP 500 and throw a generic error

Reply:
This code path is for 200, 202 and 400 responses, so this error will throw if there's an empty 200, 202 or 400 response for some reason. I couldn't get 400 to repro (and I'm not sure when we _should_ expect a 400), but I left it in this path because previously, 400 was labeled for "instance failed or terminated." If you're confident that there is no valid response with 400 then I'm happy treating 400 the same as 500. I'd still leave this error for 200 and 202 though

Reply:
Oh I see, thanks for clarifying. For safety, it might be best to keep the 400 in there then.
Just curious: if you get the status of a terminated orchestrator, do you get a 400? It would be good to confirm that comment with a test for the termination case specifically.

Given that this code path is re-used for 200, 202 and 400 - then I'd change my edit suggestion to not to include the HTTP 400 part. Would it be possible to have something **like** this for the error message instead:

```JavaScript
"DurableClient error: the Durable Functions extension replied with an empty HTTP {response.status} response."
```

Reply:
Nope, I confirmed that `Terminated` and `Failed` both return a 200 response, and I've also added tests for both of them, and updated the comments accordingly. 

Done! I've also added a try/catch statement to catch and re-throw the error from the `DurableOrchestrationStatus` constructor to follow the same pattern: `DurableClient error: ...`